### PR TITLE
AWS Signing

### PR DESCRIPTION
**Pull request by radar**
Hello! 

I've come across a usecase where I want to use this package with an AWS Elasticsearch instance, and I _also_ want to have requests signed because the documents are sensitive. I noticed that your package didn't support this yet, so I've worked on a patch to make it do that.

To make it work with AWS, I've added the ability to configure it like this:

```elixir
config :tirexs,
  uri: "uri",
  aws_elasticsearch: true,
  aws_access_key_id: "key",
  aws_secret_access_key: "secret",
  aws_region: "region"
```

If the `aws_elasticsearch` option is set to `true`, then this patch uses the `aws_auth` package to add Amazon's signature details to the URL. If it's not set, then this package continues on as it always has.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -25,7 +25,12 @@ defmodule Tirexs.Mixfile do
   end
 
   defp deps do
-    [ {:exjsx, "~> 3.2.0"}, {:ex_doc, "~> 0.12", only: :dev}, {:earmark, "~> 1.0", only: :dev} ]
+    [
+      {:exjsx, "~> 3.2.0"},
+      {:aws_auth, "~> 0.5.1"},
```
Comment (OpakAlex) [inline mix.exs]: this lib must be maybe at dev, if i don't use AWS, why this lib comes and compiles? i can add it manual, so include it just for dev and test

Comment (Zatvobor) [inline mix.exs]: agreed

**Comment by Zatvobor** [comment]
I don't like it. It should be moved away from`Tirexs.HTTP` into context specific module.

**Comment by radar** [comment]
Thanks for your consideration. Going to use my own library.

**Comment by Zatvobor** [comment]
Awesome.

**Pull request closed without merging**